src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.ts

```diff
@@ -108,65 +193,28 @@ export class OsdRecvSpeedModalComponent implements OnInit {
       };
       this.setPriority(customPriority);
     } else {
+      Object.keys(this.priorityAttrs).forEach((configOptionName) => {
+        this.osdRecvSpeedForm.get(configOptionName).reset();
+      });
       this.setPriority(this.priorities[0]);
     }
   }
 
-  getStoredPriority(callbackFn: Function) {
-    const observables = [];
-    this.priorityAttrs.forEach((configName) => {
-      observables.push(this.configService.get(configName.name));
-    });
-
-    observableForkJoin(observables)
-      .pipe(
-        mergeMap((configOptions) => {
-          const result = {};
-          configOptions.forEach((configOption) => {
-            if (configOption && 'value' in configOption) {
-              configOption.value.forEach((value) => {
-                if (value['section'] === 'osd') {
-                  result[configOption.name] = Number(value.value);
-                }
-              });
-            }
-          });
-          return of(result);
-        })
-      )
-      .subscribe((resp) => {
-        const priority = _.find(this.priorities, (p) => {
-          return _.isEqual(p.values, resp);
-        });
-
-        this.osdRecvSpeedForm.controls.customizePriority.setValue(false);
-
-        if (priority) {
-          return callbackFn(priority);
-        }
-
-        if (Object.entries(resp).length === 4) {
-          this.osdRecvSpeedForm.controls.customizePriority.setValue(true);
-          return callbackFn(Object({ name: 'custom', text: this.i18n('Custom'), values: resp }));
-        }
-
-        return callbackFn(this.priorities[0]);
-      });
-  }
-
   onPriorityChange(selectedPriorityName) {
     const selectedPriority =
       _.find(this.priorities, (p) => {
         return p.name === selectedPriorityName;
       }) || this.priorities[0];
-
     this.setPriority(selectedPriority);
```

Comment:
As discussed you should uncheck the 'Customize priority values' checkbox if you select a new profile.
```suggestion
    // Uncheck the 'Customize priority values' checkbox.
    this.osdRecvSpeedForm.get('customizePriority').setValue(false);
    // Set the priority profile values.
    this.setPriority(selectedPriority);
```